### ISSUE TITLE
[1.16.5 server] Commands executed by CustomNPC NPCs do not work

### ISSUE THREAD

**Issue by Watherfoxx** [Cycro54/Revive_Me#18]
**How to reproduce:**
- Have CustomNPC and ReviveMe
- Add a command to be executed by the NPC (when clicking on the npc, for example)
- Perform the action that would normally execute the command

**Error:**
```
[17:10:11] [Server thread/ERROR]: Exception caught during firing event: null
Index: 3
Listeners:
0: HIGH
1: ASM: class me.shedaniel.architectury.event.forge.EventHandlerImplCommon event(Lnet/minecraftforge/event/CommandEvent;)V
2: NORMAL
3: ASM: class invoker54.reviveme.common.event.CommandEvents onCommand(Lnet/minecraftforge/event/CommandEvent;)V
4: ASM: noppes.npcs.ServerEventsHandler@226485ab commandGive(Lnet/minecraftforge/event/CommandEvent;)V
5: ASM: noppes.npcs.ServerEventsHandler@226485ab commandTime(Lnet/minecraftforge/event/CommandEvent;)V
java.lang.ClassCastException

[17:06:41] [Server thread/ERROR]: Exception caught during firing event: class noppes.npcs.entity.EntityCustomNpc cannot be cast to class net.minecraft.entity.player.PlayerEntity (noppes.npcs.entity.EntityCustomNpc and net.minecraft.entity.player.PlayerEntity are in unnamed module of loader cpw.mods.modlauncher.TransformingClassLoader @b1534d3)
Index: 3
Listeners:
0: HIGH
1: ASM: class me.shedaniel.architectury.event.forge.EventHandlerImplCommon event(Lnet/minecraftforge/event/CommandEvent;)V
2: NORMAL
3: ASM: class invoker54.reviveme.common.event.CommandEvents onCommand(Lnet/minecraftforge/event/CommandEvent;)V
4: ASM: noppes.npcs.ServerEventsHandler@226485ab commandGive(Lnet/minecraftforge/event/CommandEvent;)V
5: ASM: noppes.npcs.ServerEventsHandler@226485ab commandTime(Lnet/minecraftforge/event/CommandEvent;)V
java.lang.ClassCastException: class noppes.npcs.entity.EntityCustomNpc cannot be cast to class net.minecraft.entity.player.PlayerEntity (noppes.npcs.entity.EntityCustomNpc and net.minecraft.entity.player.PlayerEntity are in unnamed module of loader cpw.mods.modlauncher.TransformingClassLoader @b1534d3)
at invoker54.reviveme.common.event.CommandEvents.onCommand(CommandEvents.java:29)
at net.minecraftforge.eventbus.ASMEventHandler_170_CommandEvents_onCommand_CommandEvent.invoke(.dynamic)
at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85)
at net.minecraftforge.eventbus.EventBus.post(EventBus.java:303)
at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
at net.minecraft.command.Commands.func_197059_a(Commands.java:260)
at noppes.npcs.NoppesUtilServer.runCommand(NoppesUtilServer.java:156)
at noppes.npcs.NoppesUtilServer.runCommand(NoppesUtilServer.java:121)
at noppes.npcs.packets.server.SPacketDialogSelected.handle(SPacketDialogSelected.java:96)
at noppes.npcs.packets.PacketServerBasic.lambda$handle$0(PacketServerBasic.java:51)
at net.minecraftforge.fml.network.NetworkEvent$Context.enqueueWork(NetworkEvent.java:215)
at noppes.npcs.packets.PacketServerBasic.handle(PacketServerBasic.java:37)
at net.minecraftforge.fml.network.simple.IndexedMessageCodec.lambda$tryDecode$3(IndexedMessageCodec.java:128)
at java.base/java.util.Optional.ifPresent(Optional.java:178)
at net.minecraftforge.fml.network.simple.IndexedMessageCodec.tryDecode(IndexedMessageCodec.java:128)
at net.minecraftforge.fml.network.simple.IndexedMessageCodec.consume(IndexedMessageCodec.java:162)
at net.minecraftforge.fml.network.simple.SimpleChannel.networkEventListener(SimpleChannel.java:80)
at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247)
at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239)
at net.minecraftforge.eventbus.EventBus.post(EventBus.java:303)
at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
at net.minecraftforge.fml.network.NetworkInstance.dispatch(NetworkInstance.java:86)
at net.minecraftforge.fml.network.NetworkHooks.lambda$onCustomPayload$1(NetworkHooks.java:91)
at java.base/java.util.Optional.map(Optional.java:260)
at net.minecraftforge.fml.network.NetworkHooks.onCustomPayload(NetworkHooks.java:91)
at net.minecraft.network.play.ServerPlayNetHandler.func_147349_a(ServerPlayNetHandler.java:2579)
at net.minecraft.network.play.client.CCustomPayloadPacket.func_148833_a(CCustomPayloadPacket.java:50)
at net.minecraft.network.play.client.CCustomPayloadPacket.func_148833_a(CCustomPayloadPacket.java:13)
at net.minecraft.network.PacketThreadUtil.func_225383_a(PacketThreadUtil.java:22)
at net.minecraft.util.concurrent.TickDelayedTask.run(SourceFile:18)
at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144)
at net.minecraft.util.concurrent.RecursiveEventLoop.func_213166_h(SourceFile:23)
at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:885)
at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:169)
at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118)
at net.minecraft.server.MinecraftServer.func_213205_aW(MinecraftServer.java:868)
at net.minecraft.server.MinecraftServer.func_213168_p(MinecraftServer.java:862)
at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:127)
at net.minecraft.server.MinecraftServer.func_213202_o(MinecraftServer.java:848)
at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:779)
at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:265)
at java.base/java.lang.Thread.run(Thread.java:840)
```

**Comment by Cycro54** [comment]
Ha, it's always something. Ty for telling me, I'll add the fix before I post the next update

**Comment by Watherfoxx** [comment]
Thanks!

**Comment by Cycro54** [comment]
I wasn't checking if the entity executing the command was actually a player, should be fixed in revive-me-1.16.5-1.6.24
